### PR TITLE
Propagate error from creating cacher and storage decorators up

### PR DESCRIPTION
**Pull request by wojtek-t**

### CODE REVIEW

File: staging/src/k8s.io/apiserver/pkg/storage/tests/cacher_test.go [hunk 1]
```diff
@@ -148,7 +149,10 @@ func updatePod(t *testing.T, s storage.Interface, obj, old *example.Pod) *exampl
 func TestGet(t *testing.T) {
 	server, etcdStorage := newEtcdTestStorage(t, etcd3testing.PathPrefix())
 	defer server.Terminate(t)
-	cacher, _ := newTestCacher(etcdStorage, 10)
+	cacher, _, err := newTestCacher(etcdStorage, 10)
+	if err != nil {
+		t.Fatalf("Couldnt' create cacher: %v", err)
```
Comment (jpbetz) [inline staging/src/k8s.io/apiserver/pkg/storage/tests/cacher_test.go]: nite: s/Couldnt'/Couldn't/ (for below instances as well)

**Review by jpbetz — COMMENTED** [review]
Thanks for doing this. One typo fix and then lgtm.

File: staging/src/k8s.io/apiserver/pkg/storage/tests/cacher_test.go (hunk 1 above)
Comment (wojtek-t) [reply staging/src/k8s.io/apiserver/pkg/storage/tests/cacher_test.go]: Done.

**Pull request merged**